sdk/src/main/scala/coop/rchain/sdk/dag/merging/ConflictResolutionLogic.scala

```diff
@@ -0,0 +1,143 @@
+          val rOpt = if (depending.nonEmpty) {
+            val newAcc         = acc -- depending
+            val newRejectedAcc = rejectedAcc ++ depending
+            (newAcc, depending, newRejectedAcc).some
+          } else none[Result]
+          rOpt.map(r => (r, r))
+      }
+      .lastOption
+      .getOrElse(init)
+    totalRejected
+  }
+
+  def clearConflictSet[D](
+      conflictScope: Set[D],
+      enforceAccept: Set[D],
+      enforceReject: Set[D],
+      conflicts: (D, D) => Boolean,
+      depends: (D, D) => Boolean,
+      seen: D => Set[D]
+  ): Set[D] = {
+    val r = conflictScope.filter { m =>
+      val rejectedSeen    = seen(m) intersect enforceReject
+      val acceptedNotSeen = enforceAccept -- seen(m)
+      rejectedSeen.exists(depends(m, _)) || acceptedNotSeen.exists(conflicts(m, _))
+    }
+    removeDependentChains(conflictScope, r, depends)
+  }
+
+  def mergingStrategy[B: Ordering, D](
+      s: MergeableScope[B, D],
+      conflicts: (D, D) => Boolean,
+      depends: (D, D) => Boolean
+  )(implicit bAsBlock: B => Block[B, D]): Either[MergeIntoFringe[B, D], MergeIntoTip[B, D]] = {
+    val seen = s.conflictScope.flatMap { b =>
+      b.deploys.map(_ -> b.seen.flatMap(_.deploys))
+    }.toMap
+    val enforceReject = clearConflictSet(
+      s.conflictSet,
+      s.acceptedFinally,
+      s.rejectedFinally,
+      conflicts,
+      depends,
+      seen
+    )
+    val tipViews = s.latestMsgs.map(bAsBlock).map { m =>
+      val mergeTarget = (m.seen.map(bAsBlock) -- s.finalScope).flatMap(_.deploys)
+      val toMerge     = mergeTarget -- enforceReject
+      val altered     = mergeTarget != toMerge
+      m.id -> (toMerge, altered)
+    }
+    // Find tips which state that does not contain deploys from enforcedRejections
+    val tipsNotAltered = tipViews.collect { case (id, (_, altered)) if !altered => id }
+    // If at least one tip state is not altered - we can merge into this tip
+    if (tipsNotAltered.nonEmpty) {
+      val viewMap      = tipViews.map { case (id, (view, _)) => (id, view) }.toMap
+      val partitioning = TipConflictResolver.tipPartitionsSorted[B, D](viewMap, tipsNotAltered)
+      MergeIntoTip(partitioning, enforceReject).asRight[MergeIntoFringe[B, D]]
+    } else
+      MergeIntoFringe(
+        s.finalityFringe,
+        tipViews.flatMap { case (_, (toMerge, _)) => toMerge },
+        enforceReject
+      ).asLeft[MergeIntoTip[B, D]]
+  }
+
+  def resolve[F[_]: Sync, B: Ordering, D: Ordering, CH, S](
+      latestMsgs: Set[B],
+      finalFringe: Set[B],
+      finalFringeStateDef: Deferred[F, S],
+      finalityState: Map[B, FinalizedDeploys[D]],
+      mergeableDiffs: Map[D, Map[CH, Long]],
+      conflicts: (D, D) => Boolean,
+      depends: (D, D) => Boolean,
+      conflicts1: (Set[D], Set[D]) => Boolean,
+      cost: D => Long,
+      state: B => S,
+      getMergeableValues: (S, CH) => F[Long]
+  )(implicit bAsBlock: B => Block[B, D]): F[ConflictResolutionResult[S, D]] =
```

Comment:
Why is `bAsBlock` so special to be implicit and the same like others like `state` or `cost`?